File: Cpp/Programming Skills/389. Find the Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        unordered_map<char, int> myMap;

        for(char c: s){
            myMap[c]++;
        }

        for(char ch: t){
            myMap[ch]--;
        }

        for(const auto& elem : myMap){
            if (elem.second == -1)
                return elem.first;
        }

        return 'a';
    }
};

File: Java/Programming Skills/389. Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        Map<Character, Integer> myMap = new HashMap<>();

        for(char c: s.toCharArray()) {
            myMap.put(c, myMap.getOrDefault(c, 0) + 1);
        }

        for(char ch: t.toCharArray()) {
            myMap.put(ch, myMap.getOrDefault(ch, 0) - 1);
        }

        for(Map.Entry<Character, Integer> entry : myMap.entrySet()) {
            if (entry.getValue() == -1) {
                return entry.getKey();
            }
        }

        return 'a'; 
    }
}
